src/site/content/en/blog/sms-otp-form/index.md

```diff
@@ -0,0 +1,256 @@
+numbers are known to be recycled and sometimes hijacked. And [the concept of OTP
+itself is not phishing resistant](https://youtu.be/kGGMgEfSzMw?t=1133).
+
+If you are looking for better security, consider using WebAuthn. You can learn
+more about it from the talk "[What's new in sign-up &
+sign-in](https://goo.gle/webauthn-video)" at the Chrome Dev Summit 2019.
+{% endAside %}
+
+## Checklist
+
+To provide the best user experience with the SMS OTP:
+
+* Use an `<input>` element with:
+    * `type="text"`
+    * `inputmode="numeric"`
+    * `autocomplete="one-time-code"`
+* Use `@${bound-domain} #${OTP}` as the last line of the SMS text message.
+* Use the Web OTP API.
+
+## Use the `<input>` element
+
+Using a form with an `<input>` element is the most important best practice you
+can follow as it will definitely work on any browser. Even if other items in
+this article don't work, the user can still enter and submit the OTP by doing it
+manually.
+
+```html
+<form action="/verify-otp" method="POST">
+  <input type="text"
+         inputmode="numeric"
+         autocomplete="one-time-code"
+         pattern="\d{6}"
+         required>
+</form>
+```
+
+The following are a few ideas to ensure an input field gets the best out of
+browser functionality.
+
+{% Aside 'gotchas' %}
+There are more general form best practices other than ones described below. [Sam
+Dutton](https://twitter.com/sw12)'s [Sign-in form best
+practices](https://web.dev/sign-in-form-best-practices/) is a great starting
+point.
+{% endAside %}
+
+### `type="text"`
+
+Because OTPs are usually 5 or 6 digit numbers, you may be tempted to use
+`type="number"` for an input field because it turns the mobile keyboard to be
+numbers only. However, it's not the best idea, because the browser expects an
+input field to be a countable number rather than a sequence of multiple numbers,
+and this causes unexpected behaviors. Using `type="number"` causes up and down
+buttons to be displayed beside the input field, and pressing these buttons
+increments or decrements the number and may remove preceding zeros.
+
+Use `type="text"` instead. This won't turn the mobile keyboard into numbers
+only, but that is fine because the next tip for using `inputmode="numeric"` does
+that job.
+
+### `inputmode="numeric"`
+
+Some websites use `type="tel"` for an input field which is smart because it
+turns the mobile keyboard to be number only (including `*` and `#`) when
+focused. Developers choose this  because, in the past, some browsers didn't
+support
+[`inputmode="numeric"`](https://developer.mozilla.org/docs/Web/HTML/Global_attributes/inputmode).
+But since [Firefox started supporting
+`inputmode="numeric"`](https://github.com/mdn/browser-compat-data/pull/6782),
+you no longer need to stick with the hack of using `type="tel"`. You can instead
+use `inputmode="numeric"` to turn the mobile keyboard into the numbers only one
+which is semantically better.
```

Comment:
```suggestion
Use `inputmode="numeric"` to change the mobile keyboard to numbers only.

Some websites use `type="tel"` for OTP input fields since it also 
turns the mobile keyboard to numbers only (including `*` and `#`) when
focused. This hack was used in the past when
[`inputmode="numeric"`](https://developer.mozilla.org/docs/Web/HTML/Global_attributes/inputmode)
wasn't widely supported. Since [Firefox started supporting
`inputmode="numeric"`](https://github.com/mdn/browser-compat-data/pull/6782),
there's no need to use the semantically incorrect `type="tel"` hack.
```